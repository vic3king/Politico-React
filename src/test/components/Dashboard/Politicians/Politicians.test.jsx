import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import PoliticiansPage from '../../../../components/Dashboard/Politicians/Politicians';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore({});

const props = {
  offices: {
    officeList: [{ id: 'id', office: 'office' }],
  },
};

const PoliticiansPageComponent = (
  <Provider store={store}>
    <PoliticiansPage {...props} />
  </Provider>
);

describe('<PoliticiansPage />', () => {
  it('renders politicians page', () => {
    const wrapper = shallow(PoliticiansPageComponent);
    expect(wrapper.find('div'));
  });
});
