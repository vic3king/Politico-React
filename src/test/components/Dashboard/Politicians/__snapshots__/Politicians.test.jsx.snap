// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoliticiansPage should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PoliticiansPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        false,
        false,
        <NavBar
          LiTagOne={
            <LiTag
              to="/"
              value="Home"
            />
          }
          LiTagTwo={
            <LiTag
              to="/logout"
              value="Logout"
            />
          }
        />,
        <div
          className="main"
        >
          <SideBar
            ButtonFour=""
            ButtonOne={
              <Button
                className="profile-btn"
                id="offic"
                onClick={[Function]}
                value="Raise Petition"
              />
            }
            ButtonThree=""
            ButtonTwo={
              <Link
                to="/results"
              >
                <Button
                  className="profile-btn"
                  id="results"
                  onClick={[Function]}
                  value="Results"
                />
              </Link>
            }
            to="results"
            user={Object {}}
          />
          <div
            className="boxed"
          >
            <section
              className="main-section"
            >
              <div>
                <h4
                  className="page-title"
                >
                  Open Positions
                  <hr />
                </h4>
              </div>
              <ProfileTopSectionCard
                handleEvent={[Function]}
                offices={
                  Array [
                    Object {
                      "name": "",
                    },
                    Object {
                      "name": "",
                    },
                  ]
                }
                value="Declare interest"
              />
            </section>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "LiTagOne": <LiTag
            to="/"
            value="Home"
          />,
          "LiTagTwo": <LiTag
            to="/logout"
            value="Logout"
          />,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <SideBar
              ButtonFour=""
              ButtonOne={
                <Button
                  className="profile-btn"
                  id="offic"
                  onClick={[Function]}
                  value="Raise Petition"
                />
              }
              ButtonThree=""
              ButtonTwo={
                <Link
                  to="/results"
                >
                  <Button
                    className="profile-btn"
                    id="results"
                    onClick={[Function]}
                    value="Results"
                  />
                </Link>
              }
              to="results"
              user={Object {}}
            />,
            <div
              className="boxed"
            >
              <section
                className="main-section"
              >
                <div>
                  <h4
                    className="page-title"
                  >
                    Open Positions
                    <hr />
                  </h4>
                </div>
                <ProfileTopSectionCard
                  handleEvent={[Function]}
                  offices={
                    Array [
                      Object {
                        "name": "",
                      },
                      Object {
                        "name": "",
                      },
                    ]
                  }
                  value="Declare interest"
                />
              </section>
            </div>,
          ],
          "className": "main",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "ButtonFour": "",
              "ButtonOne": <Button
                className="profile-btn"
                id="offic"
                onClick={[Function]}
                value="Raise Petition"
              />,
              "ButtonThree": "",
              "ButtonTwo": <Link
                to="/results"
              >
                <Button
                  className="profile-btn"
                  id="results"
                  onClick={[Function]}
                  value="Results"
                />
              </Link>,
              "to": "results",
              "user": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <section
                className="main-section"
              >
                <div>
                  <h4
                    className="page-title"
                  >
                    Open Positions
                    <hr />
                  </h4>
                </div>
                <ProfileTopSectionCard
                  handleEvent={[Function]}
                  offices={
                    Array [
                      Object {
                        "name": "",
                      },
                      Object {
                        "name": "",
                      },
                    ]
                  }
                  value="Declare interest"
                />
              </section>,
              "className": "boxed",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div>
                    <h4
                      className="page-title"
                    >
                      Open Positions
                      <hr />
                    </h4>
                  </div>,
                  <ProfileTopSectionCard
                    handleEvent={[Function]}
                    offices={
                      Array [
                        Object {
                          "name": "",
                        },
                        Object {
                          "name": "",
                        },
                      ]
                    }
                    value="Declare interest"
                  />,
                ],
                "className": "main-section",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h4
                      className="page-title"
                    >
                      Open Positions
                      <hr />
                    </h4>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Open Positions",
                        <hr />,
                      ],
                      "className": "page-title",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Open Positions",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": "hr",
                      },
                    ],
                    "type": "h4",
                  },
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "handleEvent": [Function],
                    "offices": Array [
                      Object {
                        "name": "",
                      },
                      Object {
                        "name": "",
                      },
                    ],
                    "value": "Declare interest",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "section",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          false,
          false,
          <NavBar
            LiTagOne={
              <LiTag
                to="/"
                value="Home"
              />
            }
            LiTagTwo={
              <LiTag
                to="/logout"
                value="Logout"
              />
            }
          />,
          <div
            className="main"
          >
            <SideBar
              ButtonFour=""
              ButtonOne={
                <Button
                  className="profile-btn"
                  id="offic"
                  onClick={[Function]}
                  value="Raise Petition"
                />
              }
              ButtonThree=""
              ButtonTwo={
                <Link
                  to="/results"
                >
                  <Button
                    className="profile-btn"
                    id="results"
                    onClick={[Function]}
                    value="Results"
                  />
                </Link>
              }
              to="results"
              user={Object {}}
            />
            <div
              className="boxed"
            >
              <section
                className="main-section"
              >
                <div>
                  <h4
                    className="page-title"
                  >
                    Open Positions
                    <hr />
                  </h4>
                </div>
                <ProfileTopSectionCard
                  handleEvent={[Function]}
                  offices={
                    Array [
                      Object {
                        "name": "",
                      },
                      Object {
                        "name": "",
                      },
                    ]
                  }
                  value="Declare interest"
                />
              </section>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "LiTagOne": <LiTag
              to="/"
              value="Home"
            />,
            "LiTagTwo": <LiTag
              to="/logout"
              value="Logout"
            />,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <SideBar
                ButtonFour=""
                ButtonOne={
                  <Button
                    className="profile-btn"
                    id="offic"
                    onClick={[Function]}
                    value="Raise Petition"
                  />
                }
                ButtonThree=""
                ButtonTwo={
                  <Link
                    to="/results"
                  >
                    <Button
                      className="profile-btn"
                      id="results"
                      onClick={[Function]}
                      value="Results"
                    />
                  </Link>
                }
                to="results"
                user={Object {}}
              />,
              <div
                className="boxed"
              >
                <section
                  className="main-section"
                >
                  <div>
                    <h4
                      className="page-title"
                    >
                      Open Positions
                      <hr />
                    </h4>
                  </div>
                  <ProfileTopSectionCard
                    handleEvent={[Function]}
                    offices={
                      Array [
                        Object {
                          "name": "",
                        },
                        Object {
                          "name": "",
                        },
                      ]
                    }
                    value="Declare interest"
                  />
                </section>
              </div>,
            ],
            "className": "main",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "ButtonFour": "",
                "ButtonOne": <Button
                  className="profile-btn"
                  id="offic"
                  onClick={[Function]}
                  value="Raise Petition"
                />,
                "ButtonThree": "",
                "ButtonTwo": <Link
                  to="/results"
                >
                  <Button
                    className="profile-btn"
                    id="results"
                    onClick={[Function]}
                    value="Results"
                  />
                </Link>,
                "to": "results",
                "user": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <section
                  className="main-section"
                >
                  <div>
                    <h4
                      className="page-title"
                    >
                      Open Positions
                      <hr />
                    </h4>
                  </div>
                  <ProfileTopSectionCard
                    handleEvent={[Function]}
                    offices={
                      Array [
                        Object {
                          "name": "",
                        },
                        Object {
                          "name": "",
                        },
                      ]
                    }
                    value="Declare interest"
                  />
                </section>,
                "className": "boxed",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div>
                      <h4
                        className="page-title"
                      >
                        Open Positions
                        <hr />
                      </h4>
                    </div>,
                    <ProfileTopSectionCard
                      handleEvent={[Function]}
                      offices={
                        Array [
                          Object {
                            "name": "",
                          },
                          Object {
                            "name": "",
                          },
                        ]
                      }
                      value="Declare interest"
                    />,
                  ],
                  "className": "main-section",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <h4
                        className="page-title"
                      >
                        Open Positions
                        <hr />
                      </h4>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          "Open Positions",
                          <hr />,
                        ],
                        "className": "page-title",
                      },
                      "ref": null,
                      "rendered": Array [
                        "Open Positions",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {},
                          "ref": null,
                          "rendered": null,
                          "type": "hr",
                        },
                      ],
                      "type": "h4",
                    },
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "handleEvent": [Function],
                      "offices": Array [
                        Object {
                          "name": "",
                        },
                        Object {
                          "name": "",
                        },
                      ],
                      "value": "Declare interest",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "section",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
